### ISSUE TITLE
Azure Static Web App Configuration App Settings doesn't work

### ISSUE THREAD

**Issue by wengera** [Azure/static-web-apps#1177]
**Describe the bug**

Adding app settings values to the static web app doesnt overwrite the app settings value for the deployed environments.

**To Reproduce**
1. Create App Settings Value in .env file for VueJS application
2. Use import.meta.env throughout the VueJs application to access the environment variable
3. Add the created app settings value to the azure static web app application settings for deployed environment
4. Modify the value to be different (example: different database connection string)
5. Click Save
6. The app settings value in the deployed application is not affected by how it is defined in the azure static web app configuration section

**Expected behavior**
value entered in azure static web app application settings should override what is defined in the .env file

**Screenshots**
![image](https://github.com/Azure/static-web-apps/assets/12058610/1e93b7ce-9ce0-40ae-8c3e-779112c0bbeb)

**Comment by blowsie** [comment]
You have to remember that vite is a build tool, not a run time.

Therefore variables that you set at runtime will take no effect, since the variables will have already been replaced as strings in your app at buildtime.

https://vitejs.dev/guide/env-and-mode.html#production-replacement

**Comment by humshubham** [comment]
Hey @wengera  I am facing a similar issue, were you able to find a workaround?

**Comment by markcouvaras** [comment]
I have searched everywhere for a solution but nothing seems to work. This is extremely frustrating.

**Comment by blowsie** [comment]
> I have searched everywhere for a solution but nothing seems to work. This is extremely frustrating.

Please read my response.

**Comment by markcouvaras** [comment]
Yes thank you for this it makes sense now

**Comment by thomasgauvin** [comment]
@markcouvaras, that's correct, because we don't hide the build process (you can use GitHub Actions, Azure DevOps pipelines, etc.), you must configure this yourself

**Comment by DavidDeSloovere** [comment]
> @markcouvaras, that's correct, because we don't hide the build process (you can use GitHub Actions, Azure DevOps pipelines, etc.), you must configure this yourself

Would be great if the build process was able to pull the env variables out of the Azure SWXA config though.
